mlrun/datastore/dbfs_store.py

```diff
@@ -0,0 +1,146 @@
+from mlrun.datastore.helpers import ONE_MB
+
+from .base import DataStore, FileStats
+
+# dbfs objects will be represented with the following URL: dbfs://<path>
+
+
+class DBFSStore(DataStore):
+    def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
+        super().__init__(parent, name, schema, endpoint, secrets=secrets)
+        self.get_filesystem(silent=False)
+
+    def get_filesystem(self, silent=True):
+        """return fsspec file system object, if supported"""
+        if not self._filesystem:
+            self._filesystem = fsspec.filesystem("dbfs", **self.get_storage_options())
+        return self._filesystem
+
+    def get_storage_options(self):
+        return dict(
+            token=self._get_secret_or_env("DATABRICKS_TOKEN"), instance=self.endpoint
+        )
+
+    def _prepare_path_and_verify_filesystem(self, key: str):
+        if not self._filesystem:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Performing actions on data-item without a valid filesystem"
+            )
+        if not key.startswith("/"):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Invalid key parameter - key must start with '/'"
+            )
+        return key
+
+    def get(self, key: str, size=None, offset=0) -> bytes:
+        key = self._prepare_path_and_verify_filesystem(key)
+        if size is not None and size <= 0:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "size cannot be negative or zero"
+            )
+        if not size and not offset:
+            return self._filesystem.cat_file(key)
+            # the maximum number of allowed bytes to read is 1MB
+        end = offset + (size or ONE_MB)
+        return self._filesystem._get_data(key, start=offset, end=end)
+
+    def put(self, key, data, append=False):
+
+        key = self._prepare_path_and_verify_filesystem(key)
+        if append:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Append mode not supported for Databricks file system"
+            )
+        #  can not use append mode because it overrides data.
+        mode = "w"
+        if isinstance(data, bytes):
+            mode += "b"
+        elif not isinstance(data, str):
+            raise TypeError(f"Unknown data type {type(data)}")
+        with self._filesystem.open(key, mode) as f:
+            f.write(data)
+
+    def upload(self, key: str, src_path: str):
+        key = self._prepare_path_and_verify_filesystem(key)
+        self._filesystem.put_file(src_path, key, overwrite=True)
+
+    def stat(self, key: str):
+        key = self._prepare_path_and_verify_filesystem(key)
+        files = self._filesystem.ls(key, detail=True)
+        if len(files) == 1 and files[0]["type"] == "file":
+            size = files[0]["size"]
+        elif len(files) == 1 and files[0]["type"] == "directory":
+            raise FileNotFoundError("Operation expects a file not a directory!")
```

Comment:
Here too, please no shouting at the user.

Reply:
If the user is filtering these errors? I do not think it's smart to change the convention...